packages/connector-core/src/types.ts

```diff
@@ -0,0 +1,129 @@
+import type { Language } from '@logto/phrases';
+import { Nullable } from '@silverhand/essentials';
+import { z, ZodType } from 'zod';
+
+export enum ConnectorType {
+  Email = 'Email',
+  SMS = 'SMS',
+  Social = 'Social',
+}
+
+export enum ConnectorPlatform {
+  Native = 'Native',
+  Universal = 'Universal',
+  Web = 'Web',
+}
+
+type I18nPhrases = { [Language.English]: string } & {
+  [key in Exclude<Language, Language.English>]?: string;
+};
+
+export type ConnectorMetadata = {
+  id: string;
+  target: string;
+  type: ConnectorType;
+  platform: Nullable<ConnectorPlatform>;
+  name: I18nPhrases;
+  logo: string;
+  logoDark: Nullable<string>;
+  description: I18nPhrases;
+  readme: string;
+  configTemplate: string;
+};
+
+export enum ConnectorErrorCodes {
+  General,
+  InvalidMetadata,
+  InvalidConfigGuard,
+  InsufficientRequestParameters,
+  InvalidConfig,
+  InvalidResponse,
+  TemplateNotFound,
+  NotImplemented,
+  SocialAuthCodeInvalid,
+  SocialAccessTokenInvalid,
+  SocialIdTokenInvalid,
+  AuthorizationFailed,
```

Comment:
this will be transpiled to number values right? maybe use string keys

Reply:
in the next PR?

Reply:
LOG-3987